contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Aerodrome AMO strategy
+ * @author Origin Protocol Inc
+ */
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+import { ISugarHelper } from "../../interfaces/aerodrome/ISugarHelper.sol";
+import { INonfungiblePositionManager } from "../../interfaces/aerodrome/INonfungiblePositionManager.sol";
+import { ISwapRouter } from "../../interfaces/aerodrome/ISwapRouter.sol";
+import { ICLPool } from "../../interfaces/aerodrome/ICLPool.sol";
+import { ICLGauge } from "../../interfaces/aerodrome/ICLGauge.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+
+contract AerodromeAMOStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+
+    /************************************************
+            Important (!) setup configuration
+    *************************************************/
+
+    /**
+     * In order to be able to remove a reasonable amount of complexity from the contract one of the
+     * preconditions for this contract to function correctly is to have an outside account mint a small
+     * amount of liquidity in the tick space where the contract will deploy's its liquidity and then send
+     * that NFT LP position to a dead address (transfer to zero address not allowed.) See example of such
+     * NFT LP token:
+     * https://basescan.org/token/0x827922686190790b37229fd06084350e74485b72?a=413296#inventory
+     */
+
+    /***************************************
+            Storage slot members
+    ****************************************/
+
+    /// @notice tokenId of the liquidity position
+    uint256 public tokenId;
+    /// @dev Minimum amount of tokens the strategy would be able to withdraw from the pool.
+    ///      minimum amount of tokens are withdrawn at a 1:1 price
+    uint256 public underlyingAssets;
+    /**
+     * @notice Specifies WETH to OETHb ratio the strategy contract aims for after rebalancing
+     * as 18 decimal point
+     *
+     * e.g. 0.2e18 means 20% WETH 80% OETHb
+     */
+    uint256 public poolWethShare;
```

Comment:
🟠 If the `poolWethShare` and `poolWethShareVarianceAllowed` were set together in one method, we could verify that the entire range allowed didn't go under a certain amount or over a certain amount.  This would prevent governance from setting values that would make for bad amounts of leverage. Currently, you could make 1,000,000,000,000x leverage (or worse) as governance.

Reply:
Good point, changing it to an interval here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/930402953a87f896a702f1369339f9a3a1010827